pandas/core/arrays/period.py

```diff
@@ -454,14 +792,139 @@ def _maybe_convert_timedelta(self, other):
 
+    # ------------------------------------------------------------------
+    # Formatting
+    def _format_native_types(self, na_rep=u'NaT', date_format=None,
+                             **kwargs):
+        # TODO(DatetimeArray): remove
+        values = self.astype(object)
+
+        if date_format:
+            formatter = lambda dt: dt.strftime(date_format)
+        else:
+            formatter = lambda dt: u'%s' % dt
+
+        if self.hasnans:
+            mask = self._isnan
+            values[mask] = na_rep
+            imask = ~mask
+            values[imask] = np.array([formatter(dt) for dt
+                                      in values[imask]])
+        else:
+            values = np.array([formatter(dt) for dt in values])
+        return values
+
+    def view(self, dtype=None, type=None):
+        # This is to support things like `.asi8`
+        # PeriodIndex's parent does .values.view('i8').
+        # I don't like adding this,
+        return self._data.view(dtype=dtype)
+
+    def repeat(self, repeats, *args, **kwargs):
+        """
+        Repeat elements of a Categorical.
+
+        See also
+        --------
+        numpy.ndarray.repeat
+        """
+        # TODO: Share with Categorical.repeat?
+        # need to use ndarray_values in Categorical
+        # and some kind of _constructor (from_ordinals, from_codes).
+        nv.validate_repeat(args, kwargs)
+        values = self._ndarray_values.repeat(repeats)
+        return self._from_ordinals(values, self.freq)
+
+    # Delegation...
+    def strftime(self, date_format):
+        return self._format_native_types(date_format=date_format)
+
+    def astype(self, dtype, copy=True):
+        # TODO: Figure out something better here...
+        # We have DatetimeLikeArrayMixin ->
+        #     super(...), which ends up being... DatetimeIndexOpsMixin?
+        # this is complicated.
+        # need a pandas_astype(arr, dtype).
+        from pandas import Categorical
+
+        dtype = pandas_dtype(dtype)
+
+        if is_object_dtype(dtype):
+            return np.asarray(self, dtype=object)
+        elif is_string_dtype(dtype) and not is_categorical_dtype(dtype):
+            return self._format_native_types()
+        elif is_integer_dtype(dtype):
+            return self.values.astype("i8", copy=copy)
+        elif (is_datetime_or_timedelta_dtype(dtype) and
+              not is_dtype_equal(self.dtype, dtype)) or is_float_dtype(dtype):
+            # disallow conversion between datetime/timedelta,
+            # and conversions for any datetimelike to float
+            msg = 'Cannot cast {name} to dtype {dtype}'
+            raise TypeError(msg.format(name=type(self).__name__, dtype=dtype))
+        elif is_categorical_dtype(dtype):
+            return Categorical(self, dtype=dtype)
+        elif is_period_dtype(dtype):
+            return self.asfreq(dtype.freq)
+        else:
+            return np.asarray(self, dtype=dtype)
+
+    def item(self):
```

Comment:
This is for `Index.item()` which uses `.values.item`

Reply:
This can be done in a follow-up, but things like this that are currently in `IndexOpsMixin` could go into something like `core.arrays.common.ArrayMixin`